### PR TITLE
tests: Introduce new libclient package

### PR DESCRIPTION
**Pull request by RamLavi**
### What this PR does
This PR creates a new libclient package and moves vmi related function `RunVMIAndExpectLaunch` to it.
libclient package is meant to hold functions that mainly use the go clientset in order to perform various actions on objects such as VMIs.

Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
NONE
```

### CODE REVIEW

File: tests/libclient/libclient.go [hunk 1]
```diff
@@ -18,3 +18,29 @@
...
+	"kubevirt.io/kubevirt/tests/framework/kubevirt"
+	"kubevirt.io/kubevirt/tests/libwait"
+	"kubevirt.io/kubevirt/tests/testsuite"
+)
+
+const waitingVMInstanceStart = "Waiting until the VirtualMachineInstance will start"
+
+func RunVMIAndExpectLaunch(vmi *v1.VirtualMachineInstance, timeout int) *v1.VirtualMachineInstance {
+	vmi, err := kubevirt.Client().VirtualMachineInstance(testsuite.GetTestNamespace(vmi)).Create(context.Background(), vmi, metav1.CreateOptions{})
+	Expect(err).ToNot(HaveOccurred())
```
Comment (ShellyKa13) [inline tests/libclient/libclient.go:40]: I believe its better to make this function return the error and do the expect in from where its been called.

Comment (ShellyKa13) [reply tests/libclient/libclient.go:40]: Although I see that the WaitForVMIPhase has their own Expect.. So maybe at least use:
g.ExpectWithOffset(1, err).ToNot(gomega.HaveOccurred())

File: tests/config_test.go [hunk 2]
```diff
@@ -40,6 +40,7 @@ import (
 
 	"kubevirt.io/kubevirt/pkg/config"
 	"kubevirt.io/kubevirt/pkg/libvmi"
+	"kubevirt.io/kubevirt/tests/libclient"
```
Comment (ShellyKa13) [inline tests/config_test.go:43]: nit. on most of the files put the import next to the import of kubevirt/tests

**Review by ShellyKa13 — COMMENTED** [review]
Over all great! some suggestions :)

File: tests/libclient/libclient.go (hunk 1 above)
Comment (RamLavi) [reply tests/libclient/libclient.go:40]: I like your idea, but I prefer we do this on a separate PR. 
This PR only moves a function - syntactic change, no logic.

**Review by orelmisan — COMMENTED** [review]
Thank you for the PR @RamLavi.
Please consider in-lining the `RunVMIAndExpectLaunch` instead of creating a new package for it.
I think it makes more sense because 
1. `RunVMIAndExpectLaunch` does more than one thing
2. The name of the package does not relate to its content.

File: tests/config_test.go (hunk 2 above)
Comment (fossedihelm) [reply tests/config_test.go:43]: https://github.com/kubevirt/kubevirt/blob/main/docs/coding-conventions.md#uniform-import-order-and-naming
I agree with @ShellyKa13

File: tests/libclient/libclient.go (hunk 1 above)
Comment (fossedihelm) [reply tests/libclient/libclient.go:40]: IMHO it is better to fix things while handling them.
An improvement in providing errors does not affect any readability or commit scope.
I vote for `ExpectWithOffset`.

**Review by fossedihelm — COMMENTED** [review]
Thank you @RamLavi 
I would suggest you squash the first commit since from my point of view everything can be done with only one.

File: tests/config_test.go (hunk 2 above)
Comment (RamLavi) [reply tests/config_test.go:43]: ahh I wasn't reading it right. Will fix.

File: tests/libclient/libclient.go (hunk 1 above)
Comment (RamLavi) [reply tests/libclient/libclient.go:40]: I'll add this change in a separate commit then.

Comment (ShellyKa13) [reply tests/libclient/libclient.go:40]: yeah I mentioned that WaitForVMIPhase works similarly having expect inside. So in order to still get relevant information from the calling function we can only replace: 
Expect(err).ToNot(HaveOccurred())
With:
ExpectWithOffset(1, err).ToNot(gomega.HaveOccurred())
So the function will stay the same just bubble up the error source.

File: tests/config_test.go (hunk 2 above)
Comment (fossedihelm) [reply tests/config_test.go]: @RamLavi yes, I think we need to add this enhancement in the coding convention :)

File: tests/libclient/libclient.go (hunk 1 above)
Comment (RamLavi) [reply tests/libclient/libclient.go]: Fixed. PTAL

**Comment by fossedihelm** [comment]
I think the new libclient pkg can be covered by `lint`, adding `tests/libclient/... \` in the `Makefile`

**Review by ShellyKa13 — APPROVED** [review]
/lgtm

**Pull request closed without merging**